macros/src/main/scala/com/softwaremill/macwire/Module.java

```diff
@@ -0,0 +1,73 @@
+package com.softwaremill.macwire;
+
+import java.lang.annotation.*;
+
+/**
+ * A Module groups classes of similar concern wired by macwire.
+ * Modules can then be assembled to form larger modules.
+ *
+ * A dependency on module M1 by module M2 is simply defined by a constructor parameter:
```

Comment:
Maybe also mention that modules can be combined using inheritance or abstract members?